standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -584,6 +626,27 @@ public OpenTxnsResponse openTxns(OpenTxnRequest rqst) throws MetaException {
           LOG.debug("Going to execute update <" + q + ">");
           stmt.execute(q);
         }
+
+        if (rqst.isSetReplPolicy()) {
+          List<String> rowsRepl = new ArrayList<>();
+
+          for (int i = 0; i < numTxns; i++) {
+            rowsRepl.add(
+                quoteString(rqst.getReplPolicy()) + "," + rqst.getReplSrcTxnIds().get(i) + "," + txnIds.get(i));
+          }
+
+          List<String> queriesRepl = sqlGenerator.createInsertValuesStmt(
+              "REPL_TXN_MAP (RTM_REPL_POLICY, RTM_SRC_TXN_ID, RTM_TARGET_TXN_ID)", rowsRepl);
+
+          for (String query : queriesRepl) {
+            LOG.debug("Going to execute insert <" + query + ">");
+            stmt.execute(query);
+          }
+        }
+
+        MetaStoreListenerNotifier.notifyEvent(transactionalListeners,
```

Comment:
Need to check transactionalListeners for null as notifyEvent method asserts it. Check for other callers too.

Reply:
transaction listener is crated at constructor  if its null ..so i think null check here will be redundant